crates/runix/src/flake_ref/indirect.rs

```diff
@@ -0,0 +1,118 @@
+use std::borrow::Cow;
+use std::collections::BTreeMap;
+use std::fmt::Display;
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use url::Url;
+
+use super::FlakeRefSource;
+
+/// <https://cs.github.com/NixOS/nix/blob/f225f4307662fe9a57543d0c86c28aa9fddaf0d2/src/libfetchers/path.cc#L46>
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone, PartialOrd, Ord)]
+#[serde(tag = "indirect")]
+pub struct IndirectRef {
+    pub id: String,
+    #[serde(flatten)]
+    pub attributes: BTreeMap<String, String>,
+}
+
+impl FlakeRefSource for IndirectRef {
+    fn scheme() -> Cow<'static, str> {
+        "flake".into()
+    }
+
+    fn parses(maybe_ref: &str) -> bool {
+        if maybe_ref.starts_with("flake:") {
+            return true;
+        }
+
+        if maybe_ref.contains(':') {
+            return false;
+        }
+
+        ('a'..='z').any(|prefix| maybe_ref.starts_with(prefix))
+            || ('A'..='Z').any(|prefix| maybe_ref.starts_with(prefix))
```

Comment:
```suggestion
        ('a'..='z').any(|letter| maybe_ref.starts_with(letter))
            || ('A'..='Z').any(|letter| maybe_ref.starts_with(letter))
```

Reply:
I see where youre coming from with this but I cant help to read it as "latter" which kinda confuses me reading this..

Reply:
No biggie, I just thought it was more clear since a prefix can be multiple characters and starts_with can accept a string